api/models.py

```diff
@@ -98,6 +108,34 @@ def map_data(self, fields = None):
         }
 
 
+class Nearby(object):
+    def __init__(self, db, connection):
+        self.connection = connection
+        self.db = db
+
+    def map_data(self, fields=None):
+        return {
+            'near': map(lambda station: station.map_data(include_network_id=True), self.stations)
+        }
+
+    def near(self, latitude, longitude, distance):
+        sModel = Station(self.db, self.connection)
+        self.stations = sModel.aggregate([{
+            '$geoNear': {
+                'near': {
+                    'type': 'Point',
+                    'coordinates': [
+                        latitude,
+                        longitude
+                    ]
+                },
+                'spherical': True,
+                'distanceField': 'distance',
+                'limit': 100,
+                'maxDistance': distance
```

Comment:
Is there any benefit on  using an aggregate function instead of a regular query? like:
```
return sModel.find({
    'location': {
        '$near': {
            '$geometry': {
                'type': 'Point',
                'coordinates': [longitude, latitude],
            },
            '$maxDistance': distance,
        }
    }
})
```

If there is (i have no idea), I needed to change it into the following to make it work (moving the limit into a separate aggregate)

```
self.stations = sModel.aggregate([{
            '$geoNear': {
                'near': {
                    'type': 'Point',
                    'coordinates': [
                        longitude,
                        latitude
                    ]
                },
                'spherical': True,
                'distanceField': 'distance',
                'maxDistance': distance
            },
        }, {'$limit': 100}], cursor={})
```

Reply:
The aggregate allows calculating the distance from the point you're searching from. The `distanceField` specifies in which field it should get populated in. This is especially useful when showing a list, like the one I'm doing in the Apple Watch app:

![](http://suda-dropshare.s3-eu-west-1.amazonaws.com/Simulator-Screen-Shot-Apple-Watch-Series-5-40mm-2019-09-27-at-14.53.18.png)